metroplex.js

```diff
@@ -132,23 +116,17 @@ Metroplex.readable('register', function register(address, fn) {
 Metroplex.readable('unregister', function unregister(address, fn) {
   var metroplex = this;
 
-  address = this.parse(address || metroplex.address);
+  address = metroplex.parse(address || metroplex.address);
   if (!metroplex.address) {
     if (fn) fn();
     return this;
   }
 
-  metroplex.leverage.annihilate(address, function annihilate(err) {
-    if (err) {
-      if (fn) return fn(err);
-      return metroplex.emit('error', err);
-    }
+  metroplex.redis.del(metroplex.namespace + 'server:' + metroplex.address);
```

Comment:
I would add a callback, handle the error and move the following stuff inside the callback body.